Broken BuilderBinFreeBSD in master

Example: https://github.com/ClickHouse/ClickHouse/runs/5203025719?check_suite_focus=true

Log: https://s3.amazonaws.com/clickhouse-builds/22.2/e76cf0e799bf453dcc6b12638121b47fd237efc3/binary_freebsd/build_log.log

Compilation error:

```
Feb 15 16:11:58 FAILED: src/CMakeFiles/dbms.dir/Disks/DiskLocal.cpp.o 
Feb 15 16:11:58 prlimit --as=10000000000 --data=5000000000 --cpu=1000 /usr/bin/ccache /usr/bin/clang++-13 --target=x86_64-pc-freebsd11 --sysroot=/build/cmake/freebsd/../../contrib/sysroot/freebsd-x86_64  -DBOOST_ASIO_STANDALONE=1 -DHAS_RESERVED_IDENTIFIER -DLZ4_DISABLE_DEPRECATE_WARNINGS=1 -DPOCO_ENABLE_CPP11 -DPOCO_HAVE_FD_POLL -DPOCO_OS_FAMILY_UNIX -DSNAPPY_CODEC_AVAILABLE -DSTD_EXCEPTION_HAS_STACK_TRACE=1 -DUNALIGNED_OK -DWITH_COVERAGE=0 -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -Iincludes/configs -I../src -Isrc -Isrc/Core/include -I../base/base/.. -Ibase/base/.. -I../contrib/cctz/include -I../base/pcg-random/. -I../contrib/miniselect/include -I../contrib/zstd/lib -I../contrib/lz4/lib -isystem ../contrib/libcxx/include -isystem ../contrib/libcxxabi/include -isystem ../contrib/libunwind/include -isystem ../contrib/libdivide/. -isystem ../contrib/jemalloc-cmake/include -isystem ../contrib/abseil-cpp -isystem ../contrib/croaring/cpp -isystem ../contrib/croaring/include -isystem ../contrib/cityhash102/include -isystem ../contrib/boost -isystem ../contrib/poco/Net/include -isystem ../contrib/poco/Foundation/include -isystem ../contrib/poco/NetSSL_OpenSSL/include -isystem ../contrib/poco/Crypto/include -isystem ../contrib/boringssl/include -isystem ../contrib/poco/Util/include -isystem ../contrib/poco/JSON/include -isystem ../contrib/poco/XML/include -isystem ../contrib/replxx/include -isystem ../contrib/fmtlib-cmake/../fmtlib/include -isystem ../contrib/magic_enum/include -isystem ../contrib/double-conversion -isystem ../contrib/dragonbox/include -isystem ../contrib/re2 -isystem contrib/re2-cmake -isystem ../contrib/zlib-ng -isystem contrib/zlib-ng-cmake -isystem ../contrib/pdqsort -isystem ../contrib/xz/src/liblzma/api -isystem ../contrib/azure/sdk/core/azure-core/inc -isystem ../contrib/azure/sdk/identity/azure-identity/inc -isystem ../contrib/azure/sdk/storage/azure-storage-common/inc -isystem ../contrib/azure/sdk/storage/azure-storage-blobs/inc -isystem ../contrib/snappy -isystem contrib/snappy-cmake -isystem ../contrib/msgpack-c/include -isystem ../contrib/fast_float/include -isystem ../contrib/librdkafka-cmake/include -isystem ../contrib/librdkafka/src -isystem contrib/librdkafka-cmake/auxdir -isystem ../contrib/cppkafka/include -isystem ../contrib/poco/MongoDB/include -isystem ../contrib/capnproto/c++/src -isystem ../contrib/avro/lang/c++/include -isystem ../contrib/avro/lang/c++/api -isystem ../contrib/openldap-cmake/freebsd_x86_64/include -isystem ../contrib/openldap/include -isystem ../contrib/sparsehash-c11 -isystem ../contrib/protobuf/src -isystem ../contrib/s2geometry/src -isystem ../contrib/AMQP-CPP/include -isystem ../contrib/AMQP-CPP -isystem ../contrib/libuv/include -isystem ../contrib/sqlite-amalgamation -isystem ../contrib/rocksdb/include -isystem ../contrib/libpqxx/include -isystem ../contrib/libpq -isystem ../contrib/libpq/include -isystem ../contrib/libstemmer_c/include -isystem ../contrib/wordnet-blast -isystem ../contrib/lemmagen-c/include -isystem ../contrib/simdjson/include -isystem ../contrib/rapidjson/include -isystem ../contrib/consistent-hashing -std=c++20 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=/build=. -falign-functions=32   -Wall -Wno-unused-command-line-argument  -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -Wextra -Wframe-larger-than=65536 -Wpedantic -Wno-vla-extension -Wno-zero-length-array -Wno-c11-extensions -Weverything -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c99-extensions -Wno-conversion -Wno-ctad-maybe-unsupported -Wno-deprecated-dynamic-exception-spec -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-nested-anon-types -Wno-packed -Wno-padded -Wno-shift-sign-overflow -Wno-sign-conversion -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -O2 -g -DNDEBUG -O3    -D OS_FREEBSD -nostdinc++ -std=gnu++2a -MD -MT src/CMakeFiles/dbms.dir/Disks/DiskLocal.cpp.o -MF src/CMakeFiles/dbms.dir/Disks/DiskLocal.cpp.o.d -o src/CMakeFiles/dbms.dir/Disks/DiskLocal.cpp.o -c ../src/Disks/DiskLocal.cpp
Feb 15 16:11:58 /build/src/Disks/DiskLocal.cpp:542:29: error: expected member name or ';' after declaration specifiers
Feb 15 16:11:58     constexpr static size_t PAGE_SIZE = 4096;
Feb 15 16:11:58     ~~~~~~~~~~~~~~~~~~~~~~~ ^
Feb 15 16:11:58 /build/contrib/sysroot/freebsd-x86_64/usr/include/machine/param.h:99:20: note: expanded from macro 'PAGE_SIZE'
Feb 15 16:11:58 #define PAGE_SIZE       (1<<PAGE_SHIFT) /* bytes/page */
Feb 15 16:11:58                          ^
Feb 15 16:11:58 /build/src/Disks/DiskLocal.cpp:542:29: error: expected ')'
Feb 15 16:11:58 /build/contrib/sysroot/freebsd-x86_64/usr/include/machine/param.h:99:20: note: expanded from macro 'PAGE_SIZE'
Feb 15 16:11:58 #define PAGE_SIZE       (1<<PAGE_SHIFT) /* bytes/page */
Feb 15 16:11:58                          ^
Feb 15 16:11:58 /build/src/Disks/DiskLocal.cpp:542:29: note: to match this '('
Feb 15 16:11:58 /build/contrib/sysroot/freebsd-x86_64/usr/include/machine/param.h:99:19: note: expanded from macro 'PAGE_SIZE'
Feb 15 16:11:58 #define PAGE_SIZE       (1<<PAGE_SHIFT) /* bytes/page */
Feb 15 16:11:58                         ^
Feb 15 16:11:58 /build/src/Disks/DiskLocal.cpp:560:36: error: expected unqualified-id
Feb 15 16:11:58         buf->write(data.data, data.PAGE_SIZE);
Feb 15 16:11:58                                    ^
Feb 15 16:11:58 /build/contrib/sysroot/freebsd-x86_64/usr/include/machine/param.h:99:19: note: expanded from macro 'PAGE_SIZE'
Feb 15 16:11:58 #define PAGE_SIZE       (1<<PAGE_SHIFT) /* bytes/page */
Feb 15 16:11:58                         ^
Feb 15 16:11:58 3 errors generated.
```

This code was introduced in https://github.com/ClickHouse/ClickHouse/pull/13544 but I don't see why it fails now and not in that PR

--- a/src/Disks/DiskLocal.cpp
+++ b/src/Disks/DiskLocal.cpp
@@ -539,14 +539,14 @@ catch (...)
 
 struct DiskWriteCheckData
 {
-    constexpr static size_t PAGE_SIZE = 4096;
-    char data[PAGE_SIZE]{};
+    constexpr static size_t PAGE_SIZE_IN_BYTES = 4096;
+    char data[PAGE_SIZE_IN_BYTES]{};
     DiskWriteCheckData()
     {
         static const char * magic_string = "ClickHouse disk local write check";
         static size_t magic_string_len = strlen(magic_string);
         memcpy(data, magic_string, magic_string_len);
-        memcpy(data + PAGE_SIZE - magic_string_len, magic_string, magic_string_len);
+        memcpy(data + PAGE_SIZE_IN_BYTES - magic_string_len, magic_string, magic_string_len);
     }
 };
 
@@ -557,7 +557,7 @@ try
     String tmp_template = fs::path(disk_path) / "";
     {
         auto buf = WriteBufferFromTemporaryFile::create(tmp_template);
-        buf->write(data.data, data.PAGE_SIZE);
+        buf->write(data.data, data.PAGE_SIZE_IN_BYTES);
         buf->sync();
     }
     return true;

